R/get_risktable.R

```diff
@@ -233,67 +275,74 @@ get_risktable.tidycuminc <- function(x
       default_labels = lst_stat_labels_default
     )
 
-  # named list of stats and labels
   lst_stat_labels <- as.list(label) %>% stats::setNames(statlist)
+  
+# Tidy object ---------------------------------------------------------------
 
   tidy <-
     visr_tidy_tidycuminc(x, times = times) %>%
     dplyr::select(dplyr::any_of(c("time", "outcome", "strata", names(lst_stat_labels_default))))
 
+# Create object -------------------------------------------------------------
+  
   if (isTRUE(collapse)) {
     tidy <-
       tidy %>%
       dplyr::mutate(strata = "Overall") %>%
       dplyr::group_by(dplyr::across(dplyr::any_of(c("time", "outcome", "strata")))) %>%
       dplyr::mutate(
         dplyr::across(
-          dplyr::any_of(c("n.risk", "n.event", "cumulative.event",
-                          "n.censor", "cumulative.censor")),
+          dplyr::any_of(c("n.risk", "n.event", "cum.event",
+                          "n.censor", "cum.censor")),
           ~sum(., na.rm = TRUE))
       ) %>%
       dplyr::filter(dplyr::row_number() == 1L) %>%
       dplyr::ungroup()
   }
 
-  if (group %in% "strata" || isTRUE(collapse)) {
+  if (group = "strata" || isTRUE(collapse)) {
```

Comment:
Update `group = "strata"` to `identical(group, "strata")`

Reply:
I did not want to use group %in% strata because the user can then specify group = c("strata", "statlist"). I will look at the two other options you mention.

Reply:
OK, I see. We do have a check in the code that only one is passed.

```r
  if (length(group)>1 | !(base::all(group %in% c("statlist", "strata"))))
    stop("group should equal statlist or strata.")
```

Another way you could consider checking this is with `match.arg()`. You make the default value of the argument a vector of all possible choices (e.g. `group = c("strata", "statlist")`), then at the top of the function you add this line:

```r
group <- match.arg(group)
```

If the user didn't change the argument, `group` will be assigned as `"strata"`.  The `match.arg()` checks that the user only passes a single string to the arg and ensures the value is one of `"strata"` or `"statlist"`.

Here's an example of it in use https://github.com/shannonpileggi/gtreg/blob/cc958b70ab62744a90d13723c1d45f495b092f10/R/tbl_ae.R#L105